syntax = "proto3";
package history;




import "inspection_sdk/model/inspection/host.proto";

/*
GetStatistics请求
*/
message GetStatisticsRequest {
    /*
    套件id
    */
    string pluginId = 1 ;
    /*
    巡检作业Id
    */
    string jobId = 2 ;
}

/*
GetStatistics返回
*/
message GetStatisticsResponse {
    /*
    关联的主机
    */
    inspection.InspectionHost host = 1 ;
    /*
    日志
    */
    string log = 2 ;
    /*
    巡检总项数
    */
    int32 totalNum = 3 ;
    /*
    通知项数
    */
    int32 noticeNum = 4 ;
    /*
    警告项数
    */
    int32 warningNum = 5 ;
    /*
    紧急项数
    */
    int32 emergencyNum = 6 ;
    /*
    合格项数
    */
    int32 passedNum = 7 ;
    /*
    巡检对象实例ID
    */
    string instanceId = 8 ;
    /*
    评分(精确到小数点后2位)
    */
    float score = 9 ;
    /*
    unexecuted:脚本未执行; failed:返回码大于0; normal:返回码等于0,没有异常指标; abnormal:返回码等于0,有异常指标
    */
    string status = 10 ;
}

/*
GetStatisticsApi返回
*/
message GetStatisticsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    history.GetStatisticsResponse data = 4 ;
}

